adapters/33across/33across.go

```diff
@@ -38,39 +55,45 @@ func (a *TtxAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters
 	var errs []error
 	var adapterRequests []*adapters.RequestData
 
-	adapterReq, errors := a.makeRequest(request)
-	if adapterReq != nil {
-		adapterRequests = append(adapterRequests, adapterReq)
+	// Construct request extension common to all imps
+	// NOTE: not blocking e
+	TTXRequestExt, err := makeReqExt(request)
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	// Break up multi-imp request into multiple external requests since we don't
+	// support SRA in our exchange server
+	for i := 0; i < len(request.Imp); i++ {
+		// makeRequest will either return an adapterRequest or err
+		if adapterReq, err := a.makeRequest(request, request.Imp[i], TTXRequestExt); err == nil {
+			adapterRequests = append(adapterRequests, adapterReq)
+		} else {
+			errs = append(errs, err)
+		}
 	}
-	errs = append(errs, errors...)
 
-	return adapterRequests, errors
+	return adapterRequests, errs
 }
 
-// Update the request object to include custom value
-// site.id
-func (a *TtxAdapter) makeRequest(request *openrtb.BidRequest) (*adapters.RequestData, []error) {
-	var errs []error
+func (a *TtxAdapter) makeRequest(request *openrtb.BidRequest, imp openrtb.Imp, TTXRequestExt []byte) (*adapters.RequestData, error) {
+	var imps []openrtb.Imp
 
-	// Make a copy as we don't want to change the original request
 	reqCopy := *request
```

Comment:
Nitpick: This is fine but if you are using just the copy in this method, I'd suggest passing the request to this method as value than as reference. That'd have the same effect